src/CommunityPatch/Patches/Perks/Intelligence/Engineering/HeavierSiegeEnginesPatch.cs

```diff
@@ -0,0 +1,123 @@
+      var tooltipHash = TooltipTargetMethodInfo.MakeCilSignatureSha256();
+      return hash.MatchesAnySha256(Hashes) && tooltipHash.MatchesAnySha256(TooltipHashes);
+    }
+
+    public override void Apply(Game game) {
+      var textObjStrings = TextObject.ConvertToStringList(
+        new List<TextObject> {
+          _perk.Name,
+          _perk.Description
+        }
+      );
+      // most of the properties of skills have private setters, yet Initialize is public
+      _perk.Initialize(
+        textObjStrings[0],
+        textObjStrings[1],
+        _perk.Skill,
+        (int) _perk.RequiredSkillValue,
+        _perk.AlternativePerk,
+        _perk.PrimaryRole, 20f,
+        _perk.SecondaryRole, _perk.SecondaryBonus,
+        _perk.IncrementType
+      );
+      
+      if (Applied) return;
+
+      CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo, new HarmonyMethod(PatchMethodInfo));
+      CommunityPatchSubModule.Harmony.Patch(TooltipTargetMethodInfo, postfix: new HarmonyMethod(TooltipPatchMethodInfo));
+      Applied = true;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void Prefix(ISiegeEventSide siegeEventSide, SiegeEngineType attackerEngineType, SiegeEvent.SiegeEngineConstructionProgress damagedEngine) {
+      CalculateBonusDamageAndRates(attackerEngineType, siegeEventSide, out _, out var bonusDamageOnly);
+      damagedEngine.SetHitpoints(damagedEngine.Hitpoints - bonusDamageOnly);
+    }
+    
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static void TooltipPostfix(ref List<TooltipProperty> __result, SiegeEvent.SiegeEngineConstructionProgress engineInProgress = null) {
+      var siegeEventSide = SiegeTooltipHelper.GetConstructionSiegeEventSide(engineInProgress);
+      if (siegeEventSide == null) return;
+      
+      CalculateBonusDamageAndRates(engineInProgress.SiegeEngine, siegeEventSide, out var bonusRateOnly, out var bonusDamageOnly);
+      SiegeTooltipHelper.AddPerkTooltip(__result, ActivePatch._perk, bonusRateOnly);
+      SiegeTooltipHelper.UpdateRangedDamageTooltip(__result, bonusDamageOnly);
+    }
+
+    private static void CalculateBonusDamageAndRates(
+      SiegeEngineType siegeEngineType,
+      ISiegeEventSide siegeEventSide, out float bonusRateOnly, out float bonusDamageOnly) 
+    {
+      var perk = ActivePatch._perk;
+      var baseDamage = siegeEngineType.Damage;
+      var partyMemberDamage = new ExplainedNumber(baseDamage);
+      var partyMemberRate = new ExplainedNumber(100f);
+      var parties = siegeEventSide.SiegeParties.Where(x => x.MobileParty != null);
+
+      foreach (var party in parties)
+      {
+        PerkHelper.AddPerkBonusForParty(perk, party.MobileParty, ref partyMemberRate);
+        PerkHelper.AddPerkBonusForParty(perk, party.MobileParty, ref partyMemberDamage);
+      }
+
+      bonusRateOnly = partyMemberRate.ResultNumber - 100;
+      bonusDamageOnly = partyMemberDamage.ResultNumber - baseDamage;
```

Comment:
Is this a percentage or literal `100`? If it's part of a factor, it should be 1.

Reply:
This should be OK if the expected value is a whole value.